rustls/src/server/server_conn.rs

```diff
@@ -120,13 +121,26 @@ pub trait ResolvesServerCert: Debug + Send + Sync {
     ///
     /// Return `None` to abort the handshake.
     fn resolve(&self, client_hello: ClientHello<'_>) -> Option<Arc<sign::CertifiedKey>>;
+
+    /// Return true when the server only supports raw public keys.
+    fn only_raw_public_keys(&self) -> bool {
+        false
+    }
 }
 
+/// A struct representing the server certificate types offered in the ClientHello.
+pub(super) struct ServerCertType<'a>(pub(super) &'a [CertificateType]);
```

Comment:
It's confusing that a type called `ServerCertType` (singular) contains a `[CertificateType]` (a slice of types -- plural). Which one makes sense? It doesn't seem like they can make sense at the same type.

And honestly, what's the value of introducing these newtypes in the first place?

Reply:
You're right, the client can offer multiple `CertificateType`s to the server so it should be plural. 
The added value is to make the `ClientHello::new()` signature more robust by making it harder to mix up the client and server certificate types. Do you think these ewtypes are redundant?

Reply:
I think so -- should be clear enough from the field/method context in which they appear.